File: 2019/Thang02_2018/22_02_2019/lottery.cpp
#include <iostream>
#include <stdio.h>
#include <algorithm>
#include <vector>

using namespace std;
////////

const
    int maxn=1e6+100;
int n, x, cou;
int a[maxn];
long long res;
////////

void enter()
{
    scanf("%d%d\n", &n, &x);
    for (int i=1; i<=n; i++)
        scanf("%d\n", &a[i]);
    res=(long long)(1) *n *x;
    sort(a+1, a+n+1);
}
////////

void process()
{
    cou=n+1;
    for (int i=1; i<=n; i++)
    {
        int c=upper_bound(a+1, a+n+1, a[i])-lower_bound(a+1, a+n+1, a[i]);
        if (c<cou)
        {
            x=a[i];
            cou=c;
        } else
        if (c==cou && a[i]>x) x=a[i];
    }
    res=res-(long long)(1) *x *cou;
    cout << res;
}

int main()
{
    freopen("lottery.inp", "r", stdin);
    freopen("lottery.out", "w", stdout);
    enter();
    process();
}

File: 2018/Thang10-2018/09-10-2018/lottery.pas
Const   ginp='lottery.inp';
        gout='lottery.out';
        maxn=1000010;

Var     n,gr,cr:longint;
        res:int64;
        a:array[0..maxn] of longint;

Procedure Sort(l,r:longint);
Var     i,j,t,tmp:longint;
Begin
        if l>=r then exit;
        i:=l; j:=r;
        t:=a[random(r-l+1)+l];
        repeat
                while a[i]<t do inc(i);
                while a[j]>t do dec(j);
                if i<=j then
                begin
                        tmp:=a[i]; a[i]:=a[j]; a[j]:=tmp;
                        inc(i); dec(j);
                end;
        until i>j;
        sort(l,j); sort(i,r);
End;

Procedure Enter;
Var     i:longint;
Begin
        readln(n,res);
        res:=res*n;
        for i:=1 to n do readln(a[i]);
        sort(1,n);
End;

Procedure Process;
Var     i,cou:longint;
Begin
        gr:=0; cr:=n+1; cou:=0;
        a[0]:=a[1]; a[n+1]:=a[n]+1;
        for i:=1 to n+1 do
                if a[i]=a[i-1] then inc(cou) else
                begin
                        if cou<=cr then
                        begin
                                cr:=cou;
                                gr:=a[i-1];
                        end;
                        cou:=1;
                end;
        res:=res-int64(gr)*cr;
        write(res);
End;

Begin
        Assign(input,ginp); Assign(output,gout);
        Reset(input); Rewrite(output);
        Enter;
        Process;
        Close(input); Close(output);
End.
